src/Uno.UI/UI/Xaml/Controls/Repeater/FlowLayout.cs

```diff
@@ -0,0 +1,441 @@
+			int lastRealizedItemIndex,
+			Rect lastRealizedLayoutBounds)
+		{
+			return GetExtent(
+				availableSize,
+				context,
+				firstRealized,
+				firstRealizedItemIndex,
+				firstRealizedLayoutBounds,
+				lastRealized,
+				lastRealizedItemIndex,
+				lastRealizedLayoutBounds);
+		}
+
+		void IFlowLayoutAlgorithmDelegates.Algorithm_OnElementMeasured(
+			UIElement element,
+			int index,
+			Size availableSize,
+			Size measureSize,
+			Size desiredSize,
+			Size provisionalArrangeSize,
+			VirtualizingLayoutContext context)
+		{
+			OnElementMeasured(
+				element,
+				index,
+				availableSize,
+				measureSize,
+				desiredSize,
+				provisionalArrangeSize,
+				context);
+		}
+
+		void IFlowLayoutAlgorithmDelegates.Algorithm_OnLineArranged(
+			int startIndex,
+			int countInLine,
+			double lineSize,
+			VirtualizingLayoutContext context)
+		{
+			OnLineArranged(
+				startIndex,
+				countInLine,
+				lineSize,
+				context);
+		}
+
+#endregion
+
+		void OnPropertyChanged(DependencyPropertyChangedEventArgs args)
+		{
+			var property = args.Property;
+			if (property == OrientationProperty)
+			{
+				var orientation = (Orientation)args.NewValue;
+
+				//Note: For FlowLayout Vertical Orientation means we have a Horizontal ScrollOrientation. Horizontal Orientation means we have a Vertical ScrollOrientation.
+				//i.e. the properties are the inverse of each other.
+				ScrollOrientation scrollOrientation = (orientation == Orientation.Horizontal) ? ScrollOrientation.Vertical : ScrollOrientation.Horizontal;
+				ScrollOrientation = scrollOrientation;
+			}
+			else if (property == MinColumnSpacingProperty)
+			{
+				m_minColumnSpacing = (double)args.NewValue;
+			}
+			else if (property == MinRowSpacingProperty)
+			{
+				m_minRowSpacing = (double)args.NewValue;
+			}
+			else if (property == LineAlignmentProperty)
+			{
+				m_lineAlignment = (FlowLayoutLineAlignment)args.NewValue;
+			}
+
+			InvalidateLayout();
+		}
+
+#region private helpers
+
+		double GetAverageLineInfo(
+
+			Size availableSize,
+			VirtualizingLayoutContext context,
+			FlowLayoutState flowState,
+			double avgCountInLine)
+		{
+			// default to 1 item per line with 0 size
+			double avgLineSize = 0;
+			avgCountInLine = 1;
+
+			MUX_ASSERT(context.ItemCountCore > 0);
+			if (flowState.TotalLinesMeasured == 0)
+			{
+				var tmpElement = context.GetOrCreateElementAt(0, ElementRealizationOptions.ForceCreate | ElementRealizationOptions.SuppressAutoRecycle);
+				var desiredSize = flowState.FlowAlgorithm.MeasureElement(tmpElement, 0, availableSize, context);
+				context.RecycleElement(tmpElement);
+
+				int estimatedCountInLine = Math.Max(1, (int)(Minor(availableSize) / Minor(desiredSize)));
+				flowState.OnLineArranged(0, estimatedCountInLine, Major(desiredSize), context);
+				flowState.SpecialElementDesiredSize = desiredSize;
+			}
+
+			avgCountInLine = Math.Max(1.0, flowState.TotalItemsPerLine / flowState.TotalLinesMeasured);
```

Comment:
Codacy found an issue: [Remove this useless assignment to local variable 'avgCountInLine'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5863680)